test-framework/core/src/main/java/org/keycloak/testframework/admin/KeycloakAdminClientFactory.java

```diff
@@ -0,0 +1,61 @@
+package org.keycloak.testframework.admin;
+
+import org.keycloak.OAuth2Constants;
+import org.keycloak.admin.client.Keycloak;
+import org.keycloak.admin.client.KeycloakBuilder;
+
+import java.util.ArrayList;
+import java.util.List;
+
+
+public class KeycloakAdminClientFactory {
+
+    private final List<Keycloak> adminClients = new ArrayList<>();
+
+    private final String serverUrl;
+
+    public KeycloakAdminClientFactory(String serverUrl) {
+        this.serverUrl = serverUrl;
+    }
+
+    public Keycloak create(AutoClose autoClose, String realm, String clientId, String clientSecret, String username, String password) {
```

Comment:
Since it's an enum with two options a boolean would be simpler, and maybe add it as the last argument?

```
create(String realm, String clientId, String clientSecret, String username, String password, boolean autoClose)
```